esb/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/BamMediatorExtFactory.java

```diff
@@ -32,22 +34,21 @@
 
 	public static final QName BAM_Q = new QName(SynapseConstants.SYNAPSE_NAMESPACE, "bam");
 
-	public static final String SERVER_PROFILE_LOCATION = "bamServerProfiles";
-
 	public Mediator createSpecificMediator(OMElement omElement, Properties properties) {
 		BamMediator bam = new BamMediator();
-
-		String serverProfilePath = SERVER_PROFILE_LOCATION + "/"
-				+ this.getServerProfileName(omElement);
-		String streamName = this.getStreamName(omElement);
-		String streamVersion = this.getStreamVersion(omElement);
-		if (isNotNullOrEmpty(serverProfilePath) && isNotNullOrEmpty(streamName)
-				&& isNotNullOrEmpty(streamVersion)) {
-			bam.setServerProfile(serverProfilePath);
-			bam.getStream().setStreamConfiguration(new StreamConfiguration());
-			bam.getStream().getStreamConfiguration().setName(streamName);
-			bam.getStream().getStreamConfiguration().setVersion(streamVersion);
+		// Fixing Tools-2916
+		// String serverProfilePath = SERVER_PROFILE_LOCATION + "/" + this.getServerProfileName(omElement);
+		if (StringUtils.isEmpty(this.getServerProfileName(omElement))) {
+			handleException("BAM Mediator : Server Profile Name can not be empty");
+		} else if (StringUtils.isEmpty(this.getStreamName(omElement))) {
+			handleException("BAM Mediator : Stream Name can not be empty");
+		} else if (StringUtils.isEmpty(this.getStreamVersion(omElement))) {
+			handleException("BAM Mediator : Stream Version can not be empty");
```

Comment:
Why are we not using String.isEmpty() here for checking whether string is empty?